app/include/zmk/events/behavior_state_changed.h

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2020 The ZMK Contributors
+ *
+ * SPDX-License-Identifier: MIT
+ */
+
+#pragma once
+
+#include <zephyr.h>
+#include <zmk/event_manager.h>
+#include <zmk/behavior.h>
+
+struct zmk_behavior_state_changed {
```

Comment:
This struct and the existing binding event struct are duplicative and confusing after this addition. It seems like we should be unifying this as part of this work.

Reply:
I feel `zmk_behavior_binding` is a valid type on it's own. It's a nice succinct description of the things that exist in a keymap:

```
struct zmk_behavior_binding {
    char *behavior_dev;
    uint32_t param1;
    uint32_t param2;
};
```

Reply:
On second thought, were you talking about this?

```
struct zmk_behavior_binding_event {
    int layer;
    uint32_t position;
    int64_t timestamp;
};
```

Yeah, that one and `zmk_behavior_state_changed` should definitely be unified.

Reply:
Yes exactly.